From 678cee596a136c8dd9d91ce62ae9702cb3da82b1 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 31 Oct 2024 15:46:16 +0100
Subject: [PATCH 25/25] arm64: dts: lx2160a-clearfog-cx: add description for
 retimers

SolidRun Clearfog-CX (unlike Honeycomb) has QSFP connector driven by
retimers to support long copper wires.

Add descriptions for both retimers and link them to the relevant
ethernet interfaces.

Retimers were added with board revision 1.3, earlier baords can use the
honeycomb dts.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../boot/dts/freescale/fsl-lx2160a-cex7.dtsi  |  2 +-
 .../dts/freescale/fsl-lx2160a-clearfog-cx.dts | 38 +++++++++++++++++++
 2 files changed, 39 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-lx2160a-cex7.dtsi b/arch/arm64/boot/dts/freescale/fsl-lx2160a-cex7.dtsi
index 024a119bcb7a..503e346f4d86 100644
--- a/arch/arm64/boot/dts/freescale/fsl-lx2160a-cex7.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-lx2160a-cex7.dtsi
@@ -155,7 +155,7 @@ regulator@5c {
 			};
 		};
 
-		i2c@3 {
+		i2c_smb: i2c@3 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <3>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts b/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts
index 86a9b771428d..60151fccd198 100644
--- a/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts
@@ -13,3 +13,41 @@ / {
 	compatible = "solidrun,clearfog-cx",
 		"solidrun,lx2160a-cex7", "fsl,lx2160a";
 };
+
+&dpmac3 {
+	phys = <&serdes_1 7>, <&retimer0 3>, <&retimer1 3>;
+	phy-names = "serdes", "retimer", "retimer";
+};
+
+&dpmac4 {
+	phys = <&serdes_1 6>, <&retimer0 2>, <&retimer1 2>;
+	phy-names = "serdes", "retimer", "retimer";
+};
+
+&dpmac5 {
+	phys = <&serdes_1 5>, <&retimer0 1>, <&retimer1 1>;
+	phy-names = "serdes", "retimer", "retimer";
+};
+
+&dpmac6 {
+	phys = <&serdes_1 4>, <&retimer0 0>, <&retimer1 0>;
+	phy-names = "serdes", "retimer", "retimer";
+};
+
+&i2c_smb {
+	status = "okay";
+
+	/* tx direction */
+	retimer0: retimer@22 {
+		compatible = "ti,ds250df410";
+		reg = <0x22>;
+		#phy-cells = <1>;
+	};
+
+	/* rx direction */
+	retimer1: retimer@23 {
+		compatible = "ti,ds250df410";
+		reg = <0x23>;
+		#phy-cells = <1>;
+	};
+};
-- 
2.43.0

